freyja/sample_deconv.py

```diff
@@ -25,41 +25,32 @@ def buildLineageMap():
 	return mapDict
 
 
-def build_mix_and_depth_arrays(dirName,depthDir,muts):
-	df_mix = pd.DataFrame()
-	df_depths = pd.DataFrame()
-	fnames = os.listdir(dirName)
-
-	filetype =fnames[0].split('.')[-1]
-	for fn in tqdm(os.listdir(dirName)):
-		df = pd.read_csv(dirName+fn,sep='\t')
-		try:
-			df_depth  = pd.read_csv(depthDir+fn.split(filetype)[0]+'depth',sep='\t',header=None,index_col=1)
-		except:
-			continue
-		df['mutName'] = df['REF'] + df['POS'].astype(str) + df['ALT']### this only works for substitutions, but that's what we get from the usher tree
-		df = df.drop_duplicates(subset='mutName')
-		df.set_index('mutName',inplace=True)
-		keptInds = [dfi for jdi, dfi in  enumerate(df.index) if dfi  in muts]#meh... works but should be improved
-		df_mix = df_mix.append(pd.Series({kI:df.loc[kI,'ALT_FREQ'].astype(float) for kI in keptInds},name=fn))
-		df_depths = df_depths.append(pd.Series({kI:df_depth.loc[int(re.findall(r'\d+',kI)[0]),3].astype(float) for kI in muts},name=fn))
-	return df_mix,df_depths
-
-
-def reindex_dfs(df_barcodes,df_mix,df_depths):
+def build_mix_and_depth_arrays(fn, depthFn,muts):
+	df = pd.read_csv(fn,sep='\t')
+	df_depth  = pd.read_csv(depthFn,sep='\t',header=None,index_col=1)
+	df['mutName'] = df['REF'] + df['POS'].astype(str) + df['ALT']### this only works for substitutions, but that's what we get from the usher tree
+	df = df.drop_duplicates(subset='mutName')
+	df.set_index('mutName',inplace=True)
+	keptInds = [dfi for jdi, dfi in  enumerate(df.index) if dfi  in muts]#meh... works but should be improved
+	mix = pd.Series({kI:df.loc[kI,'ALT_FREQ'].astype(float) for kI in keptInds},name=fn)
+	depths = pd.Series({kI:df_depth.loc[int(re.findall(r'\d+',kI)[0]),3].astype(float) for kI in muts},name=fn)
+	return mix,depths
+
+
+def reindex_dfs(df_barcodes,mix,depths):
 	#### reindex everything to match across the dfs
 	df_barcodes = df_barcodes.drop(index=['20A', '20C', '20D', '20H(Beta,V2)', '21C(Epsilon)'])### drop Nextstrain clade names. 
 	df_barcodes= df_barcodes.reindex(sorted(df_barcodes.columns), axis=1)
-	df_mix = df_mix.reindex(df_barcodes.columns, axis=1).fillna(0.)
+	mix = mix.reindex(df_barcodes.columns).fillna(0.)
 
-	df_depths = df_depths.drop(labels=[m0 for m0 in muts if m0 not in df_mix.columns.to_list()],axis=1)# dropping extra sequencing depth info we don't need
-	df_depths = df_depths.reindex(df_barcodes.columns, axis=1).fillna(0.)
-	return df_barcodes,df_mix,df_depths
+	depths = depths.drop(labels=[m0 for m0 in df_barcodes.columns if m0 not in mix.index.to_list()])# dropping extra sequencing depth info we don't need
```

Comment:
...this will greatly reduce look up expense. List lookups are expensive :)

```suggestion
        mix_as_set = set(mix.index)
	depths = depths.drop(labels=[m0 for m0 in df_barcodes.columns if m0 not in mix_as_set])# dropping extra sequencing depth info we don't need
```